backend/main.py

```
@@ -1005,6 +978,122 @@ async def get_user_profile(user_id: int, db: Session = Depends(get_db)):
+        }
+    except Exception as e:
+        logger.error(f"Error refreshing session: {e}")
+        return {
+            "status": "error",
+            "message": "Failed to refresh session"
+        }
+
+@app.delete("/users/session")
+async def end_user_session(
+    response: Response,
+    current_user: User = Depends(get_current_user),
```

Ruff (0.14.1)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)